nf_core/modules.py

```diff
@@ -231,3 +234,81 @@ def has_valid_pipeline(self):
         if not os.path.exists(main_nf) and not os.path.exists(nf_config):
             log.error("Could not find a main.nf or nextfow.config file in: {}".format(self.pipeline_dir))
             return False
+
+
+class ModulesTestHelper(object):
+    def __init__(self, modules_dir=""):
+        self.modules_dir = modules_dir
+
+    # Add custom dumper class to prevent overwriting the global state
+    # This prevents yaml from changing the output order
+    # See https://stackoverflow.com/a/52621703/1497385
+    class CustomDumper(yaml.Dumper):
+        def represent_dict_preserve_order(self, data):
+            return self.represent_dict(data.items())
+
+    CustomDumper.add_representer(dict, CustomDumper.represent_dict_preserve_order)
+
+    def _md5(self, fname):
+        """Generate md5 sum for file"""
+        hash_md5 = hashlib.md5()
+        with open(fname, "rb") as f:
+            for chunk in iter(lambda: f.read(4096), b""):
+                hash_md5.update(chunk)
+        md5sum = hash_md5.hexdigest()
+        return md5sum
+
+    def _get_md5_sums(self, dir, md5_sums=None):
+        """
+        Recursively go through directories and subdirectories
+        and generate tuples of (<file_path>, <md5sum>)
+        returns: list of tuples
+        """
+        if not md5_sums:
+            md5_sums = []
+        elements = glob.glob(dir + "/*")
+        for elem in elements:
+            # if file, get md5 sum
+            if os.path.isfile(elem):
+                elem_md5 = self._md5(elem)
+                md5_sums.append((elem, elem_md5))
+            # if directory, apply recursion
+            if os.path.isdir(elem):
+                md5_sums = self._get_md5_sums(elem, md5_sums)
+            else:
+                continue
+
+        return md5_sums
+
+    def generate_test_yml(self):
+        """
+        Generate the test yml file
+        """
+        # Look for output directory
+        output_dir = os.path.join(self.modules_dir, "output")
+        try:
+            assert os.path.exists(output_dir)
+            assert len(glob.glob(os.path.join(output_dir, "*"))) > 0
+        except:
+            raise FileNotFoundError("Output directory doesn't exist or is empty")
+
+        # Get list of files and their md5sums
+        md5_sums = self._get_md5_sums(output_dir)
+
+        # Create yaml output
+        file_dicts = []
+        for elem in md5_sums:
+            file_dicts.append({"path": elem[0], "md5sum": elem[1]})
```

Comment:
We're in a `Class` here, so why not just have `self.file_dicts` and add to it from the `_get_md5_sums` function directly?